rsconnect/http_support.py

```diff
@@ -55,11 +56,39 @@ def _create_ssl_connection(host_name, port, disable_tls_check, ca_data, timeout)
         )
     elif disable_tls_check:
         # noinspection PyProtectedMember
-        return http.HTTPSConnection(
-            host_name, port=(port or http.HTTPS_PORT), timeout=timeout, context=ssl._create_unverified_context(),
-        )
+        proxyURL = os.getenv("HTTPS_PROXY")
+        if proxyURL:
+            print("Using custom proxy server - TLS check disabled {}".format(proxyURL))
+            proxyURL = proxyURL.replace("http://", "")
+            if ":" in proxyURL:
+                proxyURL, proxyPort = proxyURL.split(":")
+                proxyPort = int(proxyPort)
+            else:
+                proxyPort = 8080
+            tmp = http.HTTPSConnection(
+                proxyURL, port=proxyPort, timeout=timeout, context=ssl._create_unverified_context(),
+            )
+            tmp.set_tunnel(host_name, (port or http.HTTPS_PORT))
+        else:
+            tmp = http.HTTPSConnection(
+                host_name, port=(port or http.HTTPS_PORT), timeout=timeout, context=ssl._create_unverified_context(),
+            )
+        return tmp
     else:
-        return http.HTTPSConnection(host_name, port=(port or http.HTTPS_PORT), timeout=timeout)
+        proxyURL = os.getenv("HTTPS_PROXY")
+        if proxyURL:
```

Comment:
I agree with @joncfoo - we should parse the URL. We don't want to test for falsiness, we want to test for URL validity.

Otherwise we're going to get into trouble with trailing slashes and stuff like that. The coercion to `int` below would just barf on a trailing slash and the exception would flow all the way up the stack.